forte/helpers/symmetry.cc

```diff
@@ -0,0 +1,80 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
+ * that implements a variety of quantum chemistry methods for strongly
+ * correlated electrons.
+ *
+ * Copyright (c) 2012-2020 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ * @END LICENSE
+ */
+
+#include <algorithm>
+#include <iterator>
+#include <stdexcept>
+
+#include "helpers/string_algorithms.h"
+#include "helpers/symmetry.h"
+
+namespace forte {
+
+Symmetry::Symmetry(std::string point_group) {
+    to_upper_string(point_group); // capitalize
+    // check if this is a valid point group label
+    auto search = std::find(__point_groups.cbegin(), __point_groups.cend(), point_group);
+    if (search == __point_groups.end()) {
+        std::string msg = "Point group " + point_group + " not found or supported by forte.\n";
```

Comment:
Codecoverage has a point. It may be worht doing a test that the error catching works.

Reply:
Added.